LanguageFeatures/Patterns/record_A04_t02.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// recordPattern         ::= '(' patternFields? ')'
+/// patternFields         ::= patternField ( ',' patternField )* ','?
+/// patternField          ::= ( identifier? ':' )? pattern
+/// ...
+/// It is a compile-time error if a name cannot be inferred for a named field
+/// pattern with the field name omitted
+///
+/// @description Checks that it is a compile-time error if a name cannot be
+/// inferred for a named field pattern with the field name omitted
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns,records
+
+main() {
+  var (:_) = (x1: 1);
```

Comment:
Cool! That's a counter-example to what I just said. ;-) So this test _can_ be written.